log4j-layout-jackson-json-template/src/main/java/org/apache/logging/log4j/jackson/json/template/layout/JsonTemplateLayout.java

```diff
@@ -0,0 +1,678 @@
+            case 3: return new Locale(localeFields[0], localeFields[1], localeFields[2]);
+        }
+        throw new IllegalArgumentException("invalid locale: " + locale);
+    }
+
+    @Override
+    public String toSerializable(final LogEvent event) {
+        final JsonTemplateLayoutSerializationContext context = getResetSerializationContext();
+        try {
+            encode(event, context);
+            return context.getOutputStream().toString(charset);
+        } catch (final Exception error) {
+            reloadSerializationContext(error, context);
+            throw new RuntimeException("failed serializing JSON", error);
+        }
+    }
+
+    @Override
+    public byte[] toByteArray(final LogEvent event) {
+        final JsonTemplateLayoutSerializationContext context = getResetSerializationContext();
+        try {
+            encode(event, context);
+            return context.getOutputStream().toByteArray();
+        } catch (final Exception error) {
+            reloadSerializationContext(error, context);
+            throw new RuntimeException("failed serializing JSON", error);
+        }
+    }
+
+    @Override
+    public void encode(final LogEvent event, final ByteBufferDestination destination) {
+        final JsonTemplateLayoutSerializationContext context = getResetSerializationContext();
+        try {
+            encode(event, context);
+            final ByteBuffer byteBuffer = context.getOutputStream().getByteBuffer();
+            byteBuffer.flip();
+            // noinspection SynchronizationOnLocalVariableOrMethodParameter
+            synchronized (destination) {
+                ByteBufferDestinationHelper.writeToUnsynchronized(byteBuffer, destination);
+            }
+        } catch (final Exception error) {
+            reloadSerializationContext(error, context);
+            throw new RuntimeException("failed serializing JSON", error);
+        }
+    }
+
+    // Visible for tests.
+    JsonTemplateLayoutSerializationContext getSerializationContext() {
+        return Constants.ENABLE_THREADLOCALS
+                ? serializationContextRef.get()
+                : serializationContextSupplier.get();
+    }
+
+    private JsonTemplateLayoutSerializationContext getResetSerializationContext() {
+        JsonTemplateLayoutSerializationContext context;
+        if (Constants.ENABLE_THREADLOCALS) {
+            context = serializationContextRef.get();
+            context.reset();
+        } else {
+            context = serializationContextSupplier.get();
+        }
+        return context;
+    }
+
+    private void reloadSerializationContext(
+            final Exception cause,
+            final JsonTemplateLayoutSerializationContext oldContext) {
+        try {
+            oldContext.close();
+        } catch (final Exception error) {
+            System.err.println("context close attempt suppressing the error:");
+            cause.printStackTrace(System.err);
```

Comment:
Have a better idea here rather than reaching out for `System.err`?

Reply:
The StatusLogger is likely what you want here, it's a meta-logger for reporting issues with the logging implementation.

Reply:
@carterkozak, thanks! Fixed.